apps/kg/dataloader/KGDataset.py

```diff
@@ -102,25 +104,32 @@ class KGDataset2:
 
     The triples are stored as 'head_nid\trelation_id\ttail_nid'.
     '''
-    def __init__(self, path, name):
+    def __init__(self, path, name, read_triple=True, only_train=False):
         url = 'https://s3.us-east-2.amazonaws.com/dgl.ai/dataset/{}.zip'.format(name)
 
         if not os.path.exists(os.path.join(path, name)):
             print('File not found. Downloading from', url)
             _download_and_extract(url, path, '{}.zip'.format(name))
         self.path = os.path.join(path, name)
 
-        f_ent2id = os.path.join(self.path, 'entity2id.txt')
         f_rel2id = os.path.join(self.path, 'relation2id.txt')
-
-        with open(f_ent2id) as f_ent:
-            self.n_entities = int(f_ent.readline()[:-1])
         with open(f_rel2id) as f_rel:
             self.n_relations = int(f_rel.readline()[:-1])
 
-        self.train = self.read_triple(self.path, 'train')
-        self.valid = self.read_triple(self.path, 'valid')
-        self.test = self.read_triple(self.path, 'test')
+        if only_train == True:
+            f_ent2id = os.path.join(self.path, 'local_to_global.txt')
```

Comment:
This local_to_global.txt is generated by the preprocessing progress?

Reply:
Yes, local_to_global.txt and partition_book.txt is needed to generated by preprocessing.

Reply:
Yes, local_to_global.txt and partition_book.txt is needed to be generate by preprocessing.